website/events/api/v2/views.py

```diff
@@ -193,38 +191,45 @@ def get_object(self):
         )
 
     def get(self, request, *args, **kwargs):
-        return HttpResponse(
-            content=json.dumps(
-                services.registration_fields(request, registration=self.get_object())
-            ),
+        return Response(
+            data=services.registration_fields(request, registration=self.get_object()),
             status=status.HTTP_200_OK,
         )
 
     def put(self, request, *args, **kwargs):
         original = services.registration_fields(request, registration=self.get_object())
         required_keys = set(original.keys()) - set(request.data.keys())
         if len(required_keys) > 0:
-            return HttpResponse(
-                content=f"Missing keys '{', '.join(required_keys)}' in request",
-                status=status.HTTP_400_BAD_REQUEST,
+            raise ValidationError(
+                f"Missing keys '{', '.join(required_keys)}' in request",
+                status.HTTP_400_BAD_REQUEST,
             )
+
+        if not event_permissions(
+            self.get_object().member, self.get_object().event, self.get_object().name
+        )["update_registration"]:
+            raise PermissionDenied("You cannot update this registration.")
```

Comment:
Maybe it would be better to have the `services.update_registration` throw this error. Because this check is already in there: https://github.com/svthalia/concrexit/blob/master/website/events/services.py#L205

Reply:
With that I mean catch a RegistrationError and throw a ValidationError based on it, since that can already happen currently.